prime-router/src/main/kotlin/Report.kt

```diff
@@ -200,6 +211,40 @@ class Report {
         return Report(schema, Table.create(columns), fromThisReport("deidentify"))
     }
 
+    // takes the data in the existing report and synthesizes different data from it
+    // the goal is to allow us to take real data in, move it around and scramble it so it's
+    // not able to point back to the actual records
+    fun synthesizeData(
+        synthesizeStrategies: Map<String, SynthesizeStrategy> = emptyMap(),
+        targetState: String? = null,
+        targetCounty: String? = null
+    ): Report {
+        val columns = schema.elements.map {
+            val synthesizedColumn = synthesizeStrategies[it.name]?.let { strategy ->
+                // look in the mapping parameter passed in for the current element
+                when (strategy) {
+                    // examine the synthesizeStrategy for the field
+                    // can be one of three values right now:
+                    // empty column, shuffle column, pass through column untouched
+                    SynthesizeStrategy.SHUFFLE -> {
+                        val shuffledValues = table.column(it.name).asStringColumn().shuffled()
+                        StringColumn.create(it.name, shuffledValues)
```

Comment:
I have never dealt with synthetic data generation, but I'm quite concerned that, with _small_ files, it would be possible to readily reconstruct the original data from shuffled data.

Is that a concern, and if so, how could we address it?

Reply:
It is potentially possible that this could happen. This is why I marked the one test as `@Ignore` because there were times it would fail because the potential of shuffling the results didn't move one of the data points I was checking. We could always check for some number of records threshold and below that number, skip shuffling in favor of faking data.

Reply:
I strongly support that idea of a threshold, or some other workaround for small data   Can you consider it for the next round of work?

Reply:
I'm pretty sure the Shuffle strategy (even for larger data sets) won't pass privacy scrutiny with regard to PII. Can we say that it can't be used on certain elements like phone numbers?

Reply:
Yeah we can definitely limit strategies based on the data point. Let me think about how to do that.